src/infra/dex/balancer/dto.rs

```diff
@@ -1,56 +1,243 @@
+            decimals,
         }
     }
+
+    pub fn value(&self) -> &BigDecimal {
+        &self.value
+    }
+
+    /// Convert the human readable amount to a `U256` with 18 decimals.
+    pub fn to_decimal_units(&self) -> Option<U256> {
+        big_decimal_to_u256(&(&self.value * &self.decimals))
+    }
+}
+
+impl Serialize for HumanReadableAmount {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        self.value.serialize(serializer)
+    }
+}
+
+#[serde_as]
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct Variables {
+    call_data_input: CallDataInput,
+    /// The Chain to query.
+    chain: Chain,
+    /// Whether to run `queryBatchSwap` to update the return amount with most
+    /// up-to-date on-chain values.
+    query_batch_swap: bool,
+    /// The amount to swap in human form.
+    swap_amount: HumanReadableAmount,
+    /// SwapType either exact_in or exact_out (also givenIn or givenOut).
+    swap_type: SwapType,
+    /// Token address of the tokenIn.
+    token_in: H160,
+    /// Token address of the tokenOut.
+    token_out: H160,
+    /// Which vault version to use. If none provided, will chose the better
+    /// return from either version.
+    use_protocol_version: Option<u8>,
+}
+
+/// Inputs for the call data to create the swap transaction. If this input is
+/// given, call data is added to the response.
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct CallDataInput {
+    /// How long the swap should be valid, provide a timestamp. `999999999` for
+    /// infinite. Default: infinite.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    deadline: Option<u64>,
+    /// Who receives the output amount.
+    receiver: H160,
+    /// Who sends the input amount.
+    sender: H160,
+    /// The max slippage in percent 0.01 -> 0.01%.
+    slippage_percentage: BigDecimal,
+}
+
+/// Balancer SOR API supported chains.
+#[derive(Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+enum Chain {
+    Arbitrum,
+    Avalanche,
+    Base,
+    Fantom,
+    Fraxtal,
+    Gnosis,
+    Mainnet,
+    Mode,
+    Optimism,
+    Polygon,
+    Sepolia,
+    ZkEvm,
+}
+
+impl Chain {
+    fn from_domain(chain_id: eth::ChainId) -> Result<Self, Error> {
+        match chain_id {
+            eth::ChainId::Mainnet => Ok(Self::Mainnet),
+            eth::ChainId::Gnosis => Ok(Self::Gnosis),
+            eth::ChainId::ArbitrumOne => Ok(Self::Arbitrum),
+            unsupported => Err(Error::UnsupportedChainId(unsupported)),
+        }
+    }
+}
+
+#[derive(Serialize)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+enum SwapType {
+    ExactIn,
+    ExactOut,
+}
+
+impl SwapType {
+    fn from_domain(side: order::Side) -> Self {
+        match side {
+            order::Side::Buy => Self::ExactOut,
+            order::Side::Sell => Self::ExactIn,
+        }
+    }
+}
+
+#[repr(u8)]
+enum VaultVersion {
```

Comment:
With the latest breaking change this should also be called `ProtocolVersion` to be inline with the new field name.